crates/hir-ty/src/method_resolution.rs

```diff
@@ -1459,6 +1533,7 @@ fn is_valid_fn_candidate(
 
         check_that!(table.unify(&expect_self_ty, self_ty));
 
+        let _p = tracing::span!(tracing::Level::INFO, "check_receiver_ty").entered();
```

Comment:
Wouldn't it make more sense to have these two spans inside the respective following if expressions instead of in front?

Reply:
yup, it would.